Library/brew.sh

```diff
@@ -160,6 +160,16 @@ EOS
   esac
 fi
 
+if [[ "$HOMEBREW_COMMAND" = "install" ]] && [[ -n "$HOMEBREW_DEVELOPER" ]]
+then
+  # Hide shellcheck complaint:
+  # shellcheck source=/dev/null
+  (
+    source "$HOMEBREW_LIBRARY/Homebrew/cmd/update.sh"
+    { homebrew-update --preinstall; }
```

Comment:
This line is equivalent to `homebrew-update --preinstall`. No need for the curly braces and the semicolon.

However, this is probably gonna create grief every once in a while. If the update happens to include changes to `Library/brew.sh`, then execution will continue after this line, no matter what the new code at that file offset is. In most cases this will result in an obscure error. This is exactly the reason why we have that conflated piece of code (`{ "homebrew-$HOMEBREW_COMMAND" "$@"; exit $?; }`) a few lines below, that makes sure to exit the current shell process (that started out with the old code) before leaving the already parsed piece of code in the curly braces.

A scheme that could work, but is a bit more complex: User runs `brew install <install-arguments>`. This gets transformed into `brew update --preinstall -- <install-arguments>`. Once that finishes, a new process that uses the new code needs to be spawned *from within* `brew update`, e.g. `brew install --already-updated <install-arguments>`.